test/rekt/features/kafka_source.go

```diff
@@ -561,3 +563,118 @@ func KafkaSourceWithEventAfterUpdate(kafkaSource, kafkaSink, topic string) *feat
 
 	return f
 }
+
+func KafkaSourceScalesToZeroWithKeda() *feature.Feature {
+	f := feature.NewFeatureNamed("KafkaSourceScalesToZeroWithKeda")
+
+	kafkaSource := feature.MakeRandomK8sName("kafka-source")
+	topic := feature.MakeRandomK8sName("topic")
+	kafkaSink := feature.MakeRandomK8sName("kafkaSink")
+	receiver := feature.MakeRandomK8sName("eventshub-receiver")
+	sender := feature.MakeRandomK8sName("eventshub-sender")
+
+	senderOpts := []eventshub.EventsHubOption{
+		eventshub.InputHeader("ce-specversion", cloudevents.VersionV1),
+		eventshub.InputHeader("ce-type", "com.github.pull.create"),
+		eventshub.InputHeader("ce-source", "github.com/cloudevents/spec/pull"),
+		eventshub.InputHeader("ce-subject", "123"),
+		eventshub.InputHeader("ce-id", "A234-1234-1234"),
+		eventshub.InputHeader("content-type", "application/json"),
+		eventshub.InputHeader("ce-comexampleextension1", "value"),
+		eventshub.InputHeader("ce-comexampleothervalue", "5"),
+		eventshub.InputBody(marshalJSON(map[string]string{
+			"hello": "world",
+		})),
+	}
+	matcher := AllOf(
+		HasSpecVersion(cloudevents.VersionV1),
+		HasType("com.github.pull.create"),
+		HasSource("github.com/cloudevents/spec/pull"),
+		HasSubject("123"),
+		HasId("A234-1234-1234"),
+		HasDataContentType("application/json"),
+		HasData([]byte(`{"hello":"world"}`)),
+		HasExtension("comexampleextension1", "value"),
+		HasExtension("comexampleothervalue", "5"),
+	)
```

Comment:
I think we could simplify with 

```golang
event := cetest.FullEvent()
event.SetID(uuid.New().String())
```

and then we can use this _sender_ option

```
eventshub.InputEvent(event)
```

and for matching we could just match on `event.ID()`
```golang
			assert.OnStore(receiver).MatchEvent(test.HasI(event.ID())).Exact(1))
```

since in this test we are not that interested in the "shape" of the event